src/_guides/language/sound-problems.md

```diff
@@ -369,56 +369,99 @@ the command-line Dart VM with `--strong` (where supported).
 
 ### Invalid casts
 
-In Dart 1.x `dynamic` was both a [top type][top_type_wiki] (also known as a
-_universal supertype_, or the _supertype of all other types_) and a
-[bottom type][bottom_type_wiki] (also known as the _zero_ or _empty_ type)
-depending on the context. This meant it was legal to assign, for example,
-an object of type `List<dynamic>` to a `List<String>`:
+Unlike Dart 1.x, Dart 2 enforces a sound type system. This means (roughly)
+you can't get into a situation where the value stored in some variable is
+different than the variable's static type. Like most modern statically
+typed langauges, Dart accomplishes this with a mixture of static
+(compile-time) and dynamic (runtime) checking.
 
-[top_type_wiki]: https://en.wikipedia.org/wiki/Top_type
-[bottom_type_wiki]: https://en.wikipedia.org/wiki/Bottom_type
+For example, this type error is detected at compile-time:
 
-```dart
+{:.fails-sa}
+{% prettify dart %}
 main() {
-  Map json = {'names': []};
-  List<String> names = json['names'];
+  List<int> numbers = [1, 2, 3];
+  List<String> [!string = numbers!]; // <-- Static analysis error
```

Comment:
Note that, by default, this is a warning. It is only an error if `--no-implicit-casts` is specified as a command option.

I'd suggest dropping the comment since the code is highlighted and the entire code excerpt shows "static analysis: error/warning".

Or consider changing the comment to mention the type of error:

```dart
List<String> string = numbers; // warning: invalid_assignment
```

The prose that follows the code excerpt gives the details.

Reply:
I'd prefer not to add any information around `--no-implicit-casts`, as it's not officially supported (and we aren't offering any support it).

Reply:
Oops, I just double checked: it _is_ an error even w/o the `--no-implicit-casts` (I had forgotten the `--strong` option when I did the test; w/o `--strong` it is just a warning). So the example is good as is. Apologies for the noise.